lib/validation/validators/email.ts

```diff
@@ -0,0 +1,6 @@
+/* eslint-disable-next-line no-control-regex */
+const regExp = /^(?:[A-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[A-z0-9!#$%&'*+/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0B\x0C\x0E-\x1F\x21\x23-\x5B\x5D-\x7F]|[\x01-\x09\x0B\x0C\x0E-\x7F])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9]{2,}(?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0B\x0C\x0E-\x1F\x21-\x5A\x53-\x7F]|\\[\x01-\x09\x0B\x0C\x0E-\x7F])+)\])$/i
+
+export function email(value: string): boolean {
+  return regExp.test(value)
+}
```

Comment:
The code of this validator is copied from Vuelidate:

- https://github.com/vuelidate/vuelidate/blob/08aa51495a4590c17d407a56cbbada9cf370c5d7/packages/validators/src/raw/url.js#L1-L8
- https://github.com/vuelidate/vuelidate/blob/08aa51495a4590c17d407a56cbbada9cf370c5d7/packages/validators/src/raw/core.js#L48-L56